### ISSUE TITLE
Crash in TLS socket finalizer

### ISSUE THREAD

**Issue by Leni-Vienne** [oven-sh/bun#13168]
### How can we reproduce the crash?

Call some express routes a few times. They have postgres queries for most of them, using the 'pg' package.
This crash does not happen in bun 1.20. I would like to test bun 1.21 but I am behind a company proxy so downgrading via the curl command doesn't work.
Interestingly, it says I am on Bun Canary despite the fact I just uninstalled and re-installed Bun

### Relevant log output

```shell
============================================================
Bun Canary v1.1.22-canary.96 (df33f2b2) Windows x64 (baseline)
CPU: sse42
Args: "D:\Users\leni.vienne\.bun\bin\bun.exe" "--env-file=../.env" "--hot" "run" "server.js"
Features: jsc dotenv http_server transpiler_cache(4) tsconfig(7) no_avx no_avx2
Builtins: "bun:main" "node:assert" "node:async_hooks" "node:buffer" "node:crypto" "node:dns" "node:events" "node:fs" "node:http" "node:net" "node:path" "node:querystring" "node:stream" "node:string_decoder" "node:tls" "node:tty" "node:url" "node:util" "node:util/types" "node:zlib"
Elapsed: 738032ms | User: 2406ms | Sys: 1390ms
RSS: 0.26GB | Peak: 0.26GB | Commit: 0.36GB | Faults: 149457
```

### Stack Trace (bun.report)

Bun v1.1.22 ([`df33f2b`](<https://github.com/oven-sh/bun/tree/df33f2b2a264912c7a742d0c03587082415fbba6>)) on windows x86_64_baseline [RunCommand]

Segmentation fault at address 0x2F00000057

- [`socket.c:121`](<https://github.com/oven-sh/bun/blob/df33f2b2a264912c7a742d0c03587082415fbba6/src/socket.c#L121>): `us_socket_is_closed`
- [`uws.zig:111`](<https://github.com/oven-sh/bun/blob/df33f2b2a264912c7a742d0c03587082415fbba6/src/deps/uws.zig#L111>): `isClosed`
- [`Weak.h:82`](<https://github.com/oven-sh/bun/blob/df33f2b2a264912c7a742d0c03587082415fbba6/C:/buildkite-agent/builds/windows-x64-hetzner-2/bun/bun/build/bun-webkit/include/JavaScriptCore/Weak.h#L82>): `JSC::Weak<WebCore::JSTLSSocket>::impl`
- [`HeapCell.h:57`](<https://github.com/oven-sh/bun/blob/df33f2b2a264912c7a742d0c03587082415fbba6/C:/a/WebKit/WebKit/Source/JavaScriptCore/heap/HeapCell.h#L57>): `JSC::HeapCell::zap`
- [`MarkedBlockInlines.h:476`](<https://github.com/oven-sh/bun/blob/df33f2b2a264912c7a742d0c03587082415fbba6/C:/a/WebKit/WebKit/Source/JavaScriptCore/heap/MarkedBlockInlines.h#L476>): `JSC::MarkedBlock::Handle::finishSweepKnowingHeapCellType<JSC::JSDestructibleObjectDestroyFunc>::<lambda_1>::operator()`
- [`MarkedBlockInlines.h:508`](<https://github.com/oven-sh/bun/blob/df33f2b2a264912c7a742d0c03587082415fbba6/C:/a/WebKit/WebKit/Source/JavaScriptCore/heap/MarkedBlockInlines.h#L508>): `JSC::MarkedBlock::Handle::finishSweepKnowingHeapCellType<JSC::JSDestructibleObjectDestroyFunc>`
- [`JSDestructibleObjectHeapCellType.cpp:56`](<https://github.com/oven-sh/bun/blob/df33f2b2a264912c7a742d0c03587082415fbba6/C:/a/WebKit/WebKit/Source/JavaScriptCore/runtime/JSDestructibleObjectHeapCellType.cpp#L56>): `JSC::JSDestructibleObjectHeapCellType::finishSweep`
- [`SweepingScope.h:43`](<https://github.com/oven-sh/bun/blob/df33f2b2a264912c7a742d0c03587082415fbba6/C:/a/WebKit/WebKit/Source/JavaScriptCore/heap/SweepingScope.h#L43>): `JSC::SweepingScope::~SweepingScope`
- [`FreeList.h:88`](<https://github.com/oven-sh/bun/blob/df33f2b2a264912c7a742d0c03587082415fbba6/C:/a/WebKit/WebKit/Source/JavaScriptCore/heap/FreeList.h#L88>): `JSC::FreeList::allocationWillFail`
- [`LocalAllocator.cpp:196`](<https://github.com/oven-sh/bun/blob/df33f2b2a264912c7a742d0c03587082415fbba6/C:/a/WebKit/WebKit/Source/JavaScriptCore/heap/LocalAllocator.cpp#L196>): `JSC::LocalAllocator::tryAllocateWithoutCollecting`

**Comment by Jarred-Sumner** [comment]
This crash is unrelated to express. Can you include the source code for us to reproduce it?

**Comment by Jarred-Sumner** [comment]
I think @cirospaciari bas a fix in #13176

**Comment by Jarred-Sumner** [comment]
Fixed in https://github.com/oven-sh/bun/pull/13176
